### ISSUE TITLE
conditionQualifiedByName is ignored for source="."

### ISSUE THREAD

**Issue by LeovR** [mapstruct/mapstruct#3727]
### Expected behavior

The condition specified by `conditionQualifiedByName` is used even for `source = "."`.

### Actual behavior

The condition is not used.

### Steps to reproduce the problem

```
@Mapper
public interface ConditionMapper {

    // This does not work, also not with source = "model"
    @Mapping(target = "name", source = ".", conditionQualifiedByName = "condition")
    // This works
    @Mapping(target = "name2", source = "name", conditionQualifiedByName = "condition")
    Target mapModel(Model model);

    default String map(Model value) {
        // Just a simple example for demonstration purposes.
        return value.getName();
    }

    @Condition
    @Named("condition")
    default boolean condition(Model model) {
        return false;
    }

}
```

See here for a complete example:
https://github.com/LeovR/mapstruct-conditionqualifiedbyname

### MapStruct Version

Mapstruct 1.6.2

**Comment by thunderhook** [comment]
What happens if you use `model` instead of `.`?
```java
@Mapping(target = "name", source = "model", conditionQualifiedByName = "condition")
```

Using `source = "."` is discouraged because it worked by accident and will be removed with 1.7.0, see:
* https://github.com/mapstruct/mapstruct/issues/3631

**Comment by thunderhook** [comment]
Hi @LeovR,

Thanks for the example, I just had a quick look. In order to use this condition you would need to mark it as a `@SourceParameterCondition`.
If you want it to use for both mappings, you can either mark the condition method with both annotations:
```java
@Condition
@SourceParameterCondition
@Named("condition")
default boolean condition(Model model) {
    return false;
}
```
...or use the appliesTo property introduced with 1.6.0:
```java
@Condition(appliesTo = { ConditionStrategy.PROPERTIES, ConditionStrategy.SOURCE_PARAMETERS })
@Named("condition")
default boolean condition(Model model) {
    return false;
}
```

There is some confusion about conditions and when they actually apply, which will be addressed in 1.7.0.
For more information, see:
* https://github.com/mapstruct/mapstruct/issues/3601#issuecomment-2212390504
* #3687

**Comment by LeovR** [comment]
Thank you @thunderhook,
for checking out the example and showing me the solution.
I have missed the part in the release notes regarding the change to the condition.
